### PR TITLE
docs: adding documentation for running Madara app chains with Celestia

### PR DESCRIPTION
**Pull request by SamrendraS**
## Overview
Adding Madara to Celestia docs to facilitate integrating Celestia with App Chains built on Madara.

## Checklist

- [X] New and updated code has appropriate documentation
- [X] New and updated code has new and/or updated testing
- [X] Required CI checks are passing
- [X] Visual proof for any user facing features like CLI or documentation updates
- [X] Linked issues closed with keywords

## Summary by CodeRabbit

- **New Features**
	- Added comprehensive documentation for the "Astria" section, including deployment guides and links.
	- Introduced a new "Starknet" section, featuring "Deploy to Dusknet" and "Rollups-as-a-Service" with a link to "Karnot".
	- New guides for getting started with Madara, including setup methods, deployment, and interaction instructions.
	- Added integration guidance for Celestia with Madara for blockchain applications.
	- Explained the relationship between Madara and Starknet for developers.

- **Documentation**
	- Updated links for "Sovereign SDK", "Vistara", and "Celestia Foundation Delegation Program".

### CODE REVIEW

File: .vitepress/config.ts [hunk 1]
```diff
@@ -651,6 +651,42 @@ function sidebarHome() {
                     },
                   ],
                 },
+                {
```
Comment (jcstein) [inline .vitepress/config.ts:654]: can you please alphabetize the menu in this order in the PR?

1. Astria
2. Rollkit
3. Starknet
thanks!

Comment (SamrendraS) [reply .vitepress/config.ts:654]: hi, does this work?
Astria
Dymension
Ethereum L2s:
    Arbitrum
    Optimism
    Starknet
Rollkit
Sovereign SDK
Vistara

Comment (jcstein) [reply .vitepress/config.ts:669]: works for me 😄

Comment (jcstein) [reply .vitepress/config.ts:669]: :shipit:

Comment (jcstein) [reply .vitepress/config.ts:669]: oh sorry, let's use this organization:
hi, does this work?
Astria
Dymension
Rollkit
Sovereign SDK
Vistara

Ethereum L2s:
Arbitrum
Optimism
Starknet

**Review by jcstein — CHANGES_REQUESTED** [review]

**Comment by jcstein** [comment]
also, i plan to review this in full and test next week 🫡

**Comment by jcstein** [comment]
nudging here @SamrendraS, I will plan to check back tomorrow and implement change myself if no update then

File: developers/madara-integration.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+### Launch with Celestia
+
+The following steps require the
+[Celestia CLI tool](https://docs.celestia.org/developers/node-tutorial)
+to be installed and configured
+
+### Node Initialization
+
+Run the following command to initialize a light node for the Mocha testnet:
```
Comment (jcstein) [inline developers/madara-integration.md:38]: what version is this using @SamrendraS ?

File: .vitepress/config.ts (hunk 1 above)
Comment (jcstein) [reply .vitepress/config.ts:669]: LGTM
Screenshot 2024-03-12 at 9 42 11 AM

Comment (SamrendraS) [reply .vitepress/config.ts:669]: Hi, sorry missed these notifications; will keep the order as 
Astria, Dym, EthL2, Rollkit, Sovereign, Vistara

File: developers/madara-integration.md (hunk 2 above)
Comment (SamrendraS) [reply developers/madara-integration.md:38]: i believe the version described in the node-tutorial page is used, ver="1.21.1"

Comment (SamrendraS) [reply developers/madara-integration.md:38]: we just need a celestia light node running to configure Madara with it's auth token

Comment (jcstein) [reply developers/madara-integration.md:38]: what version of celestia-node, though? 1.21.1 is the go version

Comment (SamrendraS) [reply developers/madara-integration.md:38]: hi, i'm sorry didn't realise that was the go version
the latest version that has been run with madara is 0.12.2

Comment (SamrendraS) [reply developers/madara-integration.md:38]: hi @jcstein 
can you tell me if any inputs are needed from my side

Comment (jcstein) [reply developers/madara-integration.md:38]: yes, it needs to be compatible with v0.12.4+, and ideally on v0.13.1

Comment (SamrendraS) [reply developers/madara-integration.md:38]: thanks for letting me know, will be checking the compatibility with these versions and will update the PR accordingly.

Comment (jcstein) [reply developers/madara-integration.md:38]: sweet, also please add the version number in doc, too so it is locked to a working version

Comment (jcstein) [reply developers/madara-integration.md:38]: kind bump @SamrendraS

Comment (jcstein) [reply developers/madara-integration.md:38]: hi @SamrendraS any update here?

**Pull request closed without merging**